packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -80,6 +84,8 @@ class Icon extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    assert(Directionality.of(context) != null, 'Icon widgets required an ambient Directionality.');
```

Comment:
If you're going to do this, please do it in a way that doesn't have side effects when asserts are enabled that are different than when asserts are disabled.  For example, maybe always call `Directionality.of` (both inside and outside the assert) or query whether there is a `Directionality` property without subscribing to rebuilds.

Reply:
Oh, good call. Fixed.